scripts/read-env-file.ts

```diff
@@ -0,0 +1,160 @@
+    throw new Error(`FORMAT:${reason}:${index + 1}`)
+  }
+}
+
+function makeVarsMap(lines: string[]) {
+  const map = new Map()
+  lines.forEach((line, index) => {
+    line = line.trim()
+    /* Skip empty and commented lines, preserving index for line count */
+    if (line.length === 0 || line[0] === '#') return
+
+    const splitLine = line.split('=')
+    const key = (splitLine[0] || '').trim()
+    const value = (splitLine.slice(1).join('=') || '').trim()
+
+    /* Check some error conditions and throw if met */
+    checkInvalidFormat(key, value, index)
+
+    /* If value was quoted, the asigned value should not invlude the quotes */
+    const mapValue = stripSurroundingQuotes(value)
+
+    map.set(key, mapValue)
+  })
+  return map
+}
+
+function processContent(path: string, content: string) {
+  try {
+    const lines = content.split('\n')
+    const map = makeVarsMap(lines)
+    return mapToObject(map)
+  } catch (error) {
+    const message = (error as any)?.message || ''
+    let err
+    if (message.includes('FORMAT:')) {
+      const splitMsg = message.split(':').slice(1)
+      const reason = splitMsg[0]
+      const line = parseInt(splitMsg[1], 10)
+      err = new Error(`Invalid file format (${reason}) at ${path}:${line}`)
+    } else err = error
+    throw err
+  }
+}
+
+/**
+ * Given a path to a file in .env format, returns key/value pairs in an object.
+ * @param {string} [path] The file path to read. If omitted, attempts to read from `./.env`
+ * @returns {Promise<EnvVars>} An object of corresponding key/value pairs
+ * @throws Error if file is not in valid format, or file is not found
+ */
+export async function readSingle(path?: string): Promise<EnvVars> {
+  const file = normalizePath(path)
+  return new Promise((resolve, reject) => {
+    readFile(file, { encoding: 'utf8' }, (err, content) => {
+      if (err) reject(err)
+      try {
+        resolve(processContent(file, content))
+      } catch (error) {
+        reject(error)
+      }
+    })
+  })
+}
+
+/**
+ * Given a path to a file in .env format, returns key/value pairs in an object.
+ * @param {string} [path] The file path to read. If omitted, attempts to read from `./.env`
+ * @returns {EnvVars} An object of corresponding key/value pairs
+ * @throws Error if file is not in valid format, or file is not found
+ */
+readSingle.sync = (path?: string): EnvVars => {
+  const file = normalizePath(path)
+  const content = readFileSync(file, 'utf8')
+  return processContent(file, content)
+}
+
+/**
+ * @description Given an array of paths, will read each and add the key/value
+ * pairs to an object which is returned at the end. Any re-used keys will be overwritten by
+ * the most recent value.
+ * @param {string[]} paths An array of file paths to read.
+ * @returns {Promise<EnvVars>} An aggregate object of key/value pairs
+ * @throws Error if any file is not in valid format or not found
+ */
+export async function readMultiple(paths: string[]): Promise<EnvVars> {
```

Comment:
I am not sure if we should include this method if we are not yet using it. What was your rationale to leave it?

Reply:
I was just to remove it now. In fact i think i'll put that in npm package in new repo, because its could be util in another project of 1Hive, WDYT?

I was with doubt about it, because it will avoid need use the whole package to use that function, but i think its better remove.

Reply:
Removed 👯